### PR TITLE
Removing deprecated code from io.fits.header

### LINKED ISSUE #4429: Remove AstropyDeprecationWarning from astropy.fits.header
I would leave this low hanging fruit for a newcomer: the warning in L231 in ``astropy/fits/header.py`` is around for a very long time so the next version (which is also an LTS) seems to be a good place to get rid of it finally.

What do you think @embray? 
(I run into this when testing an affiliated package where deprecations are raised as exceptions during tests, and making the code deal with KeyErrors will not help getting rid of the deprecation warning.)

### PR DESCRIPTION
**Pull request by bsipocz**
This is a rebased ad squashed version of #4445 that was milestoned as 1.2. 
I believe @embray gave the green light pending the changelog entry back in Jan, thus it would be nice to get this in the release.

Closes #4445, closes #4429 and most probably #2594, too.

cc @eteq @astrofrog

### CODE REVIEW

File: astropy/io/fits/header.py [hunk 1]
```diff
@@ -1540,9 +1528,24 @@ def remove(self, keyword):
...
+
+        all : bool, optional
+            When True, all instances of keyword will be removed.
+            Otherwise only the first instance of the given keyword is removed.
+
         """
+        keyword = Card.normalize_keyword(keyword)
+        if keyword in self._keyword_indices:
+            del self[self._keyword_indices[keyword][0]]
+            if all:
```
Comment (saimn) [inline astropy/io/fits/header.py]: `all` is maybe not a so good idea as argument name, as it shadows the builtin `all` function. Maybe `remove_all` ?

File: CHANGES.rst [hunk 2]
```diff
@@ -208,6 +208,13 @@ New Features
 
   - New function ``convenience.table_to_hdu`` to allow creating a FITS
     HDU object directly from an astropy ``Table``. [#4778]
```
Comment (saimn) [inline CHANGES.rst]: Missing newline, + wrap next lines to 80 characters ?

File: astropy/io/fits/header.py (hunk 1 above)
Comment (bsipocz) [inline astropy/io/fits/header.py]: good point, thanks!

**Comment by astrofrog** [comment]
👍

File: CHANGES.rst [hunk 3]
```diff
@@ -209,6 +209,9 @@ New Features
...
 
+  - A new optional arguments ``ignore_missing`` and ``remove_all`` are added
```
Comment (embray) [inline CHANGES.rst]: Slight typo here.  "A new optional arguments".. Should be reworded.

**Comment by embray** [comment]
Thanks @bsipocz and @Pratik151 for the original fix.

**Pull request merged**